intake/source/utils.py

```diff
@@ -0,0 +1,87 @@
+def _get_parts_of_format_string(resolved_string, literal_texts, format_specs):
```

Comment:
I gather the intent of putting the functions here would be that they could be generally useful for other file-to-dataframe plugins. Therefore, I think it's worth-while documenting the functions here, and including a section in making-plugins.html next to Remote Data.

Reply:
do you think some docs should also go in catalog? And if so where?

Reply:
Probably not, it's not general to catalogues, only used for a couple of specific plugins.
We could consider separate "spec" and "conventions" pages to describe what is required and what is likely to be found in catalog descriptions.